### PR TITLE
chore(CLI): add new CLI

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

We want to have the same behaviour from the old parser on the new CLI. To start, list commands should be the same:

- [x] List services
- [x] List fixers
- [x] List categories
- [x] List compliance
- [x] List compliance requirements
- [x] List checks json
- [x] List checks
- [x] Logging functions
- [x] Run a "mocked" prowler scan
- [x] Send findings to SH
- [x] status
- [x] output_formats
- [x] outputs_filename
- [x] output_directory
- [x] verbose
- [x] ignore_exit_code_3
- [x] no_banner
- [x] unix_timestamp
- [x] profile
- [x] Tests

Usage: `python3 cli/cli.py aws --profile x`
### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/cli.py
Comment (jfagoagas) [inline cli/cli.py:1]: I think the banner part can be removed now right?

File: cli/cli.py [hunk 1]
```diff
@@ -25,6 +35,44 @@ def list_resources(provider: str, resource_type: str):
...
+        bulk_checks_metadata = bulk_load_checks_metadata(provider)
+        checks_to_execute = load_checks_to_execute(
+            bulk_checks_metadata,
+            bulk_load_compliance_frameworks(provider),
+            None,
+            [],
+            [],
+            [],
+            [],
+            [],
```
Comment (jfagoagas) [inline cli/cli.py]: I'll add a `# TODO` here since we need to fill that in the future.

File: cli/cli.py [hunk 2]
```diff
@@ -25,6 +35,44 @@ def list_resources(provider: str, resource_type: str):
...
+def validate_frameworks(compliance_frameworks: list[str] = None):
+    if not compliance_frameworks:
+        raise typer.BadParameter("Expected at least one argument")
+    for framework in compliance_frameworks:
+        if framework not in available_compliance_frameworks:
+            print(f"{framework} is not a valid Compliance Framework\n")
+    return compliance_frameworks
 
 
 def create_list_commands(provider_typer: typer.Typer):
```
Comment (jfagoagas) [inline cli/cli.py]: The purpose of this function is to define another set of functions for Typer to get these commands, am I right? Is this a recommended pattern?

File: cli/cli.py [hunk 3]
```diff
@@ -25,6 +35,44 @@ def list_resources(provider: str, resource_type: str):
...
+
+def list_compliance_requirements(
+    provider: str, compliance_frameworks: list[str] = None
+):
+    print_compliance_requirements(
+        bulk_load_compliance_frameworks(provider), compliance_frameworks
+    )
+
+
+def validate_frameworks(compliance_frameworks: list[str] = None):
```
Comment (jfagoagas) [inline cli/cli.py]: Is this being called as a callback? Is it the way to validate CLI commands?

File: cli/cli.py (hunk 1 above)
Comment (pedrooot) [reply cli/cli.py]: Done

File: cli/cli.py
Comment (pedrooot) [reply cli/cli.py:1]: True! Done

File: cli/cli.py (hunk 3 above)
Comment (pedrooot) [reply cli/cli.py]: yes, since it's an argument the callback it's used to check the valid compliance

File: cli/cli.py (hunk 2 above)
Comment (pedrooot) [reply cli/cli.py]: It would be used this way, currently it's just to check that the compliance specified are correct

**Comment by jfagoagas** [comment]
@pedrooot remember to move `cli/tests` under `tests/cli` in the repo's root folder.

File: cli/cli.py
Comment (jfagoagas) [reply cli/cli.py:1]: remember to push it, thanks!

File: cli/cli.py (hunk 1 above)
Comment (jfagoagas) [reply cli/cli.py]: remember to push it, thanks!

**Comment by pedrooot** [comment]
Done @jfagoagas

File: cli/cli.py [hunk 4]
```diff
@@ -21,10 +31,51 @@
 
 
 def list_resources(provider: str, resource_type: str):
+    print_banner(False)
```
Comment (jfagoagas) [inline cli/cli.py]: Since this is not being done in the current CLI let's remove it.
```suggestion
```

File: cli/cli.py [hunk 5]
```diff
@@ -21,10 +31,51 @@
...
+        if resource_type == "checks":
+            print_checks(provider, sorted(checks_to_execute), bulk_checks_metadata)
+        elif resource_type == "checks-json":
+            print(list_checks_json(provider, sorted(checks_to_execute)))
+
+
+def list_compliance_requirements(
+    provider: str, compliance_frameworks: list[str] = None
+):
+    print_banner(False)
```
Comment (jfagoagas) [inline cli/cli.py]: Since this is not being done in the current CLI let's remove it.
```suggestion
```

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: cli/cli.py (hunk 4 above)
Comment (pedrooot) [reply cli/cli.py]: Done

File: cli/cli.py (hunk 5 above)
Comment (pedrooot) [reply cli/cli.py]: Done

File: tests/cli/cli_test.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    def test_list_checks_aws(self):
+        result = runner.invoke(app, ["aws", "list-checks"])
+        assert result.exit_code == 0
+        assert "available checks." in result.output
+
+    def test_list_checks_json_aws(self):
+        result = runner.invoke(app, ["aws", "list-checks-json"])
+        assert result.exit_code == 0
+        assert "aws" in result.output
+        assert result.output.startswith("{") and result.output.endswith("}\n")
```
Comment (jfagoagas) [inline tests/cli/cli_test.py]: Can you validate if this is a valid JSON following the schema?

Comment (pedrooot) [reply tests/cli/cli_test.py]: Done

**Comment by jfagoagas** [comment]
We have decided to close this PR, we are not going to work on this at this time. We will get back to it in the future.

**Pull request closed without merging**